specs/agents-versioning-publishing.md

```diff
@@ -0,0 +1,62 @@
+# Hypertrace Agents versioning and publishing
+
+This document describes versioning and publishing requirements for Hypertrace agents.
+All Hypertrace agents follow these requirements:
+
+* Semantic versioning for all released artifacts.
+* Release every merge commit into the main or maintenance branch as `"snapshot"`.
+* Maintenance branches for older versions should be named as `release-<major>.<minor>` e.g. `release-1.0`.
```

Comment:
this is not a great convention, as it can be confused with release itself. It is more typical to use N.N.x as a branch name. HEAD branches should not be persistent, typically, so it isn't likely such a convention will conflict with anything.

Reply:
> this is not a great convention, as it can be confused with release itself.

It would be in zipkin since there the releases are triggered by `release-<>.<>.<>` tag. I haven't seen this used in Hypertrace but we can certainly use the `N.N.x`. 

> It is more typical to use N.N.x as a branch name.

In what organizations? 

* https://github.com/kubernetes/kubernetes/branches uses `release-`
* https://github.com/prometheus/prometheus/branches uses `release-`
* https://github.com/apache/spark/branches uses `branch-`
* https://github.com/apache/cassandra/branches uses `cassandra-`
* https://github.com/tensorflow/tensorflow/branches uses `r.x.x`
* https://github.com/facebook/react/branches uses `x.x-stable`

The name `release-x.x` would be obviously confusing if we were triggering release by `release-x.x.x` tag. I am not sure if there are any plans to do so.

Reply:
spring, zipkin, kafka

I think the release-X thing is a CNCF'ism